#! /bin/sh /usr/share/dpatch/dpatch-run
## chan-modem.dpatch by  Simon Peter <dn.tlp@gmx.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: chan_modem.c should set default read & write format

@DPATCH@
diff -urNad --exclude=CVS --exclude=.svn ./channels/chan_modem.c /tmp/dpep-work.H6ITFA/asterisk-1.0.9.dfsg/channels/chan_modem.c
--- ./channels/chan_modem.c	2004-10-24 22:15:23.000000000 +0100
+++ /tmp/dpep-work.H6ITFA/asterisk-1.0.9.dfsg/channels/chan_modem.c	2005-08-09 21:33:36.000000000 +0100
@@ -524,12 +524,15 @@
 struct ast_channel *ast_modem_new(struct ast_modem_pvt *i, int state)
 {
 	struct ast_channel *tmp;
+	int fmt = ast_best_codec(i->mc->formats);
 	tmp = ast_channel_alloc(1);
 	if (tmp) {
 		snprintf(tmp->name, sizeof(tmp->name), "Modem[%s]/%s", i->mc->name, i->dev + 5);
 		tmp->type = type;
 		tmp->fds[0] = i->fd;
 		tmp->nativeformats = i->mc->formats;
+		tmp->readformat = fmt;
+		tmp->writeformat = fmt;
 		ast_setstate(tmp, state);
 		if (state == AST_STATE_RING)
 			tmp->rings = 1;
